cmd/influx-tools/importer/importer.go

```diff
@@ -0,0 +1,186 @@
+package importer
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/influxdb/cmd/influx-tools/server"
+	"github.com/influxdata/influxdb/services/meta"
+	"github.com/influxdata/influxdb/tsdb"
+	"github.com/influxdata/influxdb/tsdb/engine/tsm1"
+	"go.uber.org/zap"
+)
+
+type Config struct {
+	Database string
+	Data     tsdb.Config
+}
+
+type Importer struct {
+	MetaClient server.MetaClient
+	db         string
+	dataDir    string
+
+	rpi       *meta.RetentionPolicyInfo
+	log       *zap.Logger
+	sh        *shardWriter
+	sfile     *tsdb.SeriesFile
+	sw        *seriesWriter
+	buildTsi  bool
+	seriesBuf []byte
+}
+
+const seriesBatchSize = 1000
+
+func NewImporter(client server.MetaClient, db string, dataDir string, buildTsi bool, log *zap.Logger) *Importer {
+	i := &Importer{MetaClient: client, db: db, dataDir: dataDir, buildTsi: buildTsi, log: log}
+	if !buildTsi {
+		i.seriesBuf = make([]byte, 0, 2048)
+	}
+	return i
+}
+
+func (i *Importer) Close() error {
+	defer i.CloseShardGroup()
+	err := i.closeSeriesFile()
+	if err != nil {
+		return err
+	}
+	return i.CloseShardGroup()
+}
```

Comment:
This will call `CloseShardGroup` twice under normal conditions. It would be better to bring in `ErrorList`(https://github.com/influxdata/ingen/blob/master/errorlist.go) and do something like the following, by improving the `ErrorList` type to clean up the `if err != nil` dance:

```go
func (i *Importer) Close() error {
	var errs ErrorList
	errs.Add(i.closeSeriesFile()) // does nothing if `err == nil`
	errs.Add(i.CloseShardGroup())
	return errs.Err() // returns nil if len(errs) == 0
}
```